datafusion/substrait/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -628,32 +660,6 @@ async fn qualified_catalog_schema_table_reference() -> Result<()> {
     roundtrip("SELECT a,b,c,d,e FROM datafusion.public.data;").await
 }
 
-#[tokio::test]
```

Comment:
I believe these tests are fully replaced by the stricter `roundtrip_self_join` tests I added above.

These tests worked before since they hit the case (1) - the tables have overlapping names but they're both qualified, so the `DFSchema::check_names()` doesn't mind, and they don't read those overlapping columns in the final result. 

They would pass with the changes here as well, they'd just have additional SubqueryAliases.